### PR TITLE
feat(datagrid): update ag-grid

### PR DESCRIPTION
**Pull request by jare-talend**
**What is the problem this PR is trying to solve?**
we use ag-grid#20

**What is the chosen solution to this problem?**
upgrade ag-grid to 20.2

Before, we use forceRedrawRows to refresh the grid after props changed on componentDidUpdate. But, this method refresh the alls grids.
Now, it is not necessary, ag-grid detects if a cell have to be refreshed when the data in rowData changes their reference.
Optimize the datagrid with pass your rowData to Ag-grid to avoid a long parsing with the				method getRowData very cost with a lot of rows and some loops in addation. If you don't use rowData, you have one loop in your application, again a loop with the getRowData and a loop in ag-grid to detect the changes. With rowData, we can avoid the loop with the getRowData

**Before**: 
`
<Datagrid forceRedrawRows={() => true} getRowDataFn={()=>rowsData} />
`
**After:**
`
<Datagrid getRowDataFn={()=>rowsData}/>
`

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

[ ] **This PR introduces a breaking change**

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -11905,15 +11905,6 @@ prop-types-extra@^1.0.1:
     react-is "^16.3.2"
     warning "^3.0.0"
 
-prop-types@15.6.0:
```
Comment (jmfrancois) [inline yarn.lock]: 👍

File: packages/datagrid/src/components/DefaultCellRenderer/DefaultCellRenderer.component.js [hunk 2]
```diff
@@ -12,18 +12,23 @@ import theme from './DefaultCell.scss';
 
 export const CELL_RENDERER_COMPONENT = 'cellRenderer';
 
-export default function DefaultCellRenderer({ avroRenderer, colDef, value, getComponent, data }) {
+function DefaultCellRenderer({ avroRenderer, colDef, value, getComponent, data }) {
 	let content;
 
+	let plainValue = value;
+	if (value.toJS) {
+		plainValue = value.toJS();
```
Comment (jsomsanith-tlnd) [inline packages/datagrid/src/components/DefaultCellRenderer/DefaultCellRenderer.component.js]: Please memoize that

File: packages/datagrid/stories/datagrid.component.js [hunk 3]
```diff
@@ -175,4 +177,41 @@ storiesOf('Component Datagrid', module)
...
 	})
-	.add('dynamic change data', () => <DynamicDataGrid />);
+	.add('@deprecated - dynamic change data with forceRedrawRows', () => (
+		<DynamicDataGrid forceRedraw />
+	))
+	.add('dynamic change data', () => <DynamicDataGrid />)
+	.add('faster datagrid', () => (
+		<div style={{ height: '100vh' }}>
+			<IconsProvider />
+			<p>
```
Comment (jsomsanith-tlnd) [inline packages/datagrid/stories/datagrid.component.js]: All this description is an explanation of the implementation. Remove that from the story please, and add it to the PR description please

File: packages/datagrid/src/components/DefaultCellRenderer/DefaultCellRenderer.component.js [hunk 4]
```diff
@@ -12,13 +13,20 @@ import theme from './DefaultCell.scss';
...
 
+function convertValue(value) {
+	if (!value.toJS) {
+		return value;
+	}
+
+	return value.toJS();
+}
+
+const memoizedConvertValue = memoize(convertValue);
```
Comment (jsomsanith-tlnd) [inline packages/datagrid/src/components/DefaultCellRenderer/DefaultCellRenderer.component.js]: You never clear the cache, I think you you use a memoize-one solution.
Here it's a functional component, you can't attach a memoized function by instance, but there is the React.useMemo hook 🎉

Comment (jare-talend) [reply packages/datagrid/src/components/DefaultCellRenderer/DefaultCellRenderer.component.js]: We have decided that use react.memo isn't interesting. ag-grid use a virtualization so the row is destroy or add when the user scrolls. So, we lost the benefit of react.memo when the row is destroyed.

**Pull request merged**